scripts/server.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+'''
+An HTTP server that proxies requests to solr to search for similar incipits.
+
+Run a debug server for development with:
+    $ export FLASK_APP=scripts/server.py FLASK_ENV=development
+    $ flask run
+
+'''
+import os
+
+from flask import Flask, g, jsonify, render_template, request
+from parasolr.query import SolrQuerySet
+from parasolr.solr.client import SolrClient
+
+from scripts import __version__
+
+# read settings from environment on startup
+SOLR_URL = os.getenv('SOLR_URL', 'http://localhost:8983/solr/')
+SOLR_CORE = os.getenv('SOLR_CORE', 'pemm')
+
+# create a new flask app from this module
+app = Flask(__name__)
+
+
+@app.route('/')
+def root():
+    '''Display version info and a basic html search form.'''
+    return render_template('index.html', version=__version__)
+
+
+@app.route('/search', methods=['GET', 'POST'])
+def search():
+    '''Search for an incipit and return a list of matching results.'''
+    if request.method == 'POST':
+        search_term = request.form['incipit']
+        output_format = request.form.get('format', '')
+    elif request.method == 'GET':
+        search_term = request.args.get('incipit', '')
+        output_format = request.args.get('format', '')
+
+    queryset = SolrQuerySet(get_solr())
+    if search_term:
+        queryset = queryset.highlight('incipit_txt_gez', q=search_term,
+                                      method='unified', fragsize=0) \
+            .search(incipit_txt_gez=search_term) \
+            .order_by('-score') \
+            .only('*', 'score')
```

Comment:
what does this `only()` do? I remember you explaining it to me once but I'm not sure what it's doing here

Reply:
It adjusts the fields we want returned from Solr. I'm using it here to include the relevance score in the results, but we probably should limit to just the fields we need for the sheets lookup. (Incipit, macomber id, source, relevance?)

Reply:
cool, thx for explaining. those 4 sound good to me